products-built-on-oxen/lokinet/guides/lokinet-on-a-soho-router.md

```diff
@@ -0,0 +1,325 @@
+
+`# systemctl edit lokinet`
+
+this will enter an editor. add the following snippet affer the the comment block where it tells you in the editor:
+
+```ini
+[Service]
+PostExecStart=+/sbin/iptables -A POSTROUTING -j MASQUERADE -o lokinet0
+```
+
+then save and quit that editor (in nano it's control + o then enter then control + x).
+
+### hostapd
+
+this daemon manages the wireless access point.
+
+#### configuration
+
+in this example we set the ssid of the wifi network to be `lokinet` with the very strong password of `lokipass` (for real networks use a better password).
+you MUST also set your country code, for the usa it's `US` for australia it is `AU` and for germany it is `DE`.
+
+see the debian wiki page on [hostapd](https://wiki.debian.org/hostap) for more help on configuring.
+
+config file at `/etc/hostapd/hostapd.conf`:
+
+```ini
+interface=wifi0
+hw_mode=g
+channel=0
+ieee80211d=1
+ieee80211n=1
+ieee80211ac=1         
+wmm_enabled=1
+auth_algs=1
+wpa=2
+wpa_key_mgmt=WPA-PSK
+rsn_pairwise=CCMP
+
+ssid=lokinet
+wpa_passphrase=lokipass
+
+;; country_code=[insert your country code here]
+
+```
+
+#### caveats
+
+many settings of hostapd may need to change depending on your hardware and region.
+you will likely need to consult the main documentation for hostapd if you have more troubles.
+
+### unbound
+
+this daemon does dns for ICANN (the normal internet) via an encrypted DNS tunnel.
+
+#### configuration
+
+in this configuration example we use [adguard's DNS over TLS](https://adguard-dns.io/en/public-dns.html) as our dns provider. we cache results locally, and force ipv4 only.
+
+
+this file goes to `/etc/unbound/unbound.conf.d/wifi.conf`
+
+```conf
+server:
+    port: 5533
+    interface: 127.0.42.2
+    access-control: 127.0.0.0/16 allow
+    do-ip6: no
+    do-ip4: yes
+    tls-system-cert: yes
+    tls-upstream: yes
+    rrset-roundrobin: yes
+    qname-minimisation-strict: yes
+    hide-identity: yes
+    hide-version: yes
+    aggressive-nsec: yes
+    cache-max-ttl: 14400
+    cache-min-ttl: 1200
+    rrset-cache-size: 256m
+    msg-cache-size: 128m
+    
+forward-zone:
+    name: "."
+    forward-tls-upstream: yes
+    forward-addr: 94.140.15.15@853#dns.adguard-dns.com
+    
+```
+
+### dnsmasq
+
+this daemon handles dhcp on the wifi network.
+
+#### configuration
+
+add to the end of `/etc/dnsmasq.conf` the following line:
+
+```conf
+conf-dir=/etc/dnsmasq.d/,*.conf
+```
+
+then add the main configuration we'll put it at `/etc/dnsmasq.d/wifi.conf`
+
+```conf
+
+# dont load /etc/resolv.conf
+no-resolv
+
+# dns provider for lokinet
+server=/loki/127.0.42.1#5533
+server=/snode/127.0.42.1#5533
+server=/100.10.in-addr.arpa/127.0.42.1#5533
+
+# icann dns provider via local unbound daemon
+server=/./127.0.42.2#5533
+
+# use the interface called wifi0 for all our dns and dhcp
+interface=wifi0
+
+# dont use these interfaces, specifically adding eth0 so we dont take over the upstream network.
+except-interface=lo,lokinet0,eth0
```

Comment:
This line is unnecessary: `interface=wifi0` is already not listening on eth0, etc.